### PR TITLE
[RADOS] Test suite reduction part-1

### PR DESCRIPTION
**Pull request by harshkumarRH**
Jira tracker : [RHCEPHQE-16832](https://issues.redhat.com/browse/RHCEPHQE-16832)

This PR consolidates and reduces the number of test suites being executed for RADOS, goal is to have fewer cluster deployments and combine the workflows to reduce the instances created in CI. 

The changes have been made in Squid and Reef for now to observe pipeline improvements.
Next follow-up PR should include the same changes and further reduction if possible for Pacific and Quincy

Signed-off-by: Harsh Kumar <hakumar@redhat.com>

### CODE REVIEW

**Comment by pdhiran** [comment]
metadata MR : https://gitlab.cee.redhat.com/rhcs-qe/cephqe-jenkins/-/merge_requests/751

File: suites/reef/rados/tier-2_rados_test-stretch-mode.yaml [hunk 1]
```diff
@@ -1,231 +0,0 @@
...
-      module: test_stretch_deployment_with_placement.py
-      polarion-id: CEPH-83573621
-      config:
-        no_affinity: false
-        stretch_rule_name: stretch_rule
-        tiebreaker_mon_site_name: tiebreaker
-      desc: Enables connectivity mode and deploys cluster with Stretch rule with tiebreaker node
-      abort-on-fail: true
-
-  - test:
```
Comment (s-vipin) [inline suites/reef/rados/tier-2_rados_test-stretch-mode.yaml:178]: All tests from **suites/reef/rados/tier-2_rados_test-stretch-mode.yaml** are removed from reef suites and not moved to other suite. Is this expected?

Comment (harshkumarRH) [reply suites/reef/rados/tier-2_rados_test-stretch-mode.yaml:178]: `tier-2_rados_test-stretch-mode.yaml` was a subset of `tier-3_rados_test-location-stretch-mode.yaml` and therefore completely removed.
However, `Mon election strategies in stretch mode` test is not part of `tier-3_rados_test-location-stretch-mode.yaml`, very good catch

**Review by pdhiran — APPROVED** [review]
Looks good

**Pull request merged**